src/components/EraserLink/index.tsx

```diff
@@ -0,0 +1,81 @@
+import { Link } from "@solidjs/router"
+import { ParentProps } from "solid-js"
+import "./eraserlink.css"
+
+const ERASER_TRACKING_PARAMS = ""
+
+type EraserLinkData = {
+	workspaceId: string
+	elementsId?: string
+}
+
+export const getEraserLinkData = (href: string): EraserLinkData | void => {
+	const matches = /app.eraser.io\/workspace\/(\w+)(.*elements=(\w+))?/.exec(
+		href
+	)
+
+	if (!matches) {
+		return
+	}
+
+	if (matches[3]) {
+		return {
+			workspaceId: matches[1],
+			elementsId: matches[3],
+		}
+	}
+	return {
+		workspaceId: matches[1],
+	}
+}
+
+const EraserLink = ({
+	linkData,
+	children,
+}: ParentProps<{
+	linkData: EraserLinkData
+}>) => {
+	const workspaceUrl = `https://app.eraser.io/workspace/${linkData.workspaceId}`
+	const elementParams = linkData.elementsId
+		? `elements=${linkData.elementsId}`
+		: ""
+
+	const linkUrl = elementParams
+		? `${workspaceUrl}?${elementParams}&${ERASER_TRACKING_PARAMS}`
+		: `${workspaceUrl}?${ERASER_TRACKING_PARAMS}`
+
+	// if there are no children or this was a right click-copy as markdown embed.
+	if (
+		children === undefined ||
+		(Array.isArray(children) && children[0] === "View on Eraser")
```

Comment:
if the first child is "View on Eraser" it assumes this is a markdown embed from Eraser.